### PR TITLE
Refactoring of blending classes into a module and separating out the …

### PR DESCRIPTION
**Pull request by fionaRust**
…Triangular Blending across adjacent points

#189 

Refactoring ahead of adding new functionality: 
* Moved all blending files to a lib/blending directory
* Moved TriangularBlendingAcrossAdjacentPoints to own directory.

Testing:
 - [x] Ran tests and they passed OK
 - [ ] Added new tests for the new feature(s) - N/A

### CODE REVIEW

**Comment by fionaRust** [comment]
Happy to get feedback on names of files.

File: lib/improver/blending/blend_across_adjacent_points.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
```
Comment (bayliffe) [inline lib/improver/blending/blend_across_adjacent_points.py]: I guess this filename is to accommodate different functions for the adjacent weighting beyond just triangular?
e.g. the exponential decay suggested in Nigel's document

File: lib/improver/blending/blend_across_adjacent_points.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+from improver.blending.weighted_blend import WeightedBlendAcrossWholeDimension
+
+
+class TriangularWeightedBlendAcrossAdjacentPoints(object):
+    """
+    Apply a Weighted blend to a coordinate, using triangular weights at each
+    point in the coordinate.
+    Returns a cube with the same coordinates as the input cube, with each
+    point in the coordinate of interest having been blended with the adjacent
+    points according to a triangular weighting
```
Comment (bayliffe) [inline lib/improver/blending/blend_across_adjacent_points.py]: This doesn't need to be on a new line.

File: lib/improver/blending/blend_across_adjacent_points.py (hunk 1 above)
Comment (fionaRust) [inline lib/improver/blending/blend_across_adjacent_points.py]: Yep, I think we might want to use slightly different weighting functions in here, so I left it open for that.

File: lib/improver/blending/blend_across_adjacent_points.py (hunk 2 above)
Comment (fionaRust) [inline lib/improver/blending/blend_across_adjacent_points.py]: Fixed

**Pull request merged**